### PR TITLE
feat(worker): Add make gltf

### PR DESCRIPTION
**Pull request by dongri**
# Overview
Add writing processing in gltf format to the action-sink.

## What I've done
Add gltf format

## Memo
Reference
* https://github.com/reearth/plateau-gis-converter/blob/main/nusamai-gltf/examples/make_gltf.rs
* https://github.com/MIERUNE/plateau-gis-converter/tree/main/nusamai/src/sink/gltf

## Summary by CodeRabbit

- **New Features**
  - Added support for GLTF file format, enhancing 3D content management capabilities.
  - Introduced a new workflow for converting CityGML data into GLTF format, facilitating interoperability.

- **Improvements**
  - Enhanced texture, material, and image handling for GLTF exports, improving representation and robustness.
  
- **Bug Fixes**
  - Addressed issues in existing file formats and ensured compliance with new GLTF specifications.

### CODE REVIEW

File: worker/crates/action-sink/src/file/material.rs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+#[derive(Debug, Serialize, Clone, PartialEq, Deserialize)]
+pub struct Material {
+    pub base_color: [f32; 4],
+    pub base_texture: Option<Texture>,
+    // NOTE: Adjust the hash implementation if you add more fields
+}
+
+impl Eq for Material {}
+
+impl Hash for Material {
```
Comment (miseyu) [inline worker/crates/action-sink/src/file/material.rs:20]: This is the type of
to be used. If the following code needs modification, please modify the following code.
https://github.com/reearth/reearth-flow/blob/main/worker/crates/types/src/geometry.rs#L265-L268

File: worker/crates/action-sink/src/file/material.rs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                base_color_texture: tex,
+                ..Default::default()
+            }),
+            ..Default::default()
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Clone, Hash, PartialEq, Eq, Deserialize)]
+pub struct Texture {
```
Comment (miseyu) [inline worker/crates/action-sink/src/file/material.rs:56]: This is the type of
to be used. If the following code needs modification, please modify the following code.
https://github.com/reearth/reearth-flow/blob/main/worker/crates/types/src/geometry.rs#L248-L251

File: worker/crates/action-sink/src/file/material.rs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        });
+        nusamai_gltf_json::Texture {
+            source: Some(image_index as u32),
+            ..Default::default()
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Clone, Hash, PartialEq, Eq, Deserialize)]
+pub struct Image {
```
Comment (miseyu) [inline worker/crates/action-sink/src/file/material.rs:76]: This code should be added here.
https://github.com/reearth/reearth-flow/blob/main/worker/crates/types/src/geometry.rs

**Comment by miseyu** [comment]
@dongri 
Commented on.
Please check it.

File: worker/crates/action-sink/src/file/material.rs (hunk 1 above)
Comment (dongri) [reply worker/crates/action-sink/src/file/material.rs]: Move to geometry.rs

File: worker/crates/action-sink/src/file/material.rs (hunk 2 above)
Comment (dongri) [reply worker/crates/action-sink/src/file/material.rs]: Move to geometry.rs

File: worker/crates/action-sink/src/file/material.rs (hunk 3 above)
Comment (dongri) [reply worker/crates/action-sink/src/file/material.rs]: Move to geometry.rs

File: worker/Cargo.toml [hunk 4]
```diff
@@ -57,6 +57,7 @@ nusamai-citygml = {git = "https://github.com/reearth/plateau-gis-converter", tag
 nusamai-geometry = {git = "https://github.com/reearth/plateau-gis-converter", tag = "v0.0.1-rc2", features = ["serde"]}
 nusamai-plateau = {git = "https://github.com/reearth/plateau-gis-converter", tag = "v0.0.1-rc2", features = ["serde"]}
 nusamai-projection = {git = "https://github.com/reearth/plateau-gis-converter", tag = "v0.0.1-rc2"}
+nusamai-gltf = {git = "https://github.com/reearth/plateau-gis-converter", branch = "exports-nusamai-gltf"}
```
Comment (miseyu) [inline worker/Cargo.toml:60]: We have imported this branch and release it.
https://github.com/reearth/plateau-gis-converter/releases/tag/v0.0.1-rc3
```suggestion
nusamai-gltf = {git = "https://github.com/reearth/plateau-gis-converter", tag="v0.0.1-rc3"}
```

**Review by miseyu — APPROVED** [review]
Can you please just correct the crate git tag?
LGTM for implementation!

File: worker/crates/action-sink/src/file/gltf.rs [hunk 5]
```diff
@@ -0,0 +1,417 @@
...
+        .into(),
+        extensions_used: vec![
+            "EXT_mesh_features".to_string(),
+            "EXT_structural_metadata".to_string(),
+            "EXT_texture_webp".to_string(),
+        ],
+        ..Default::default()
+    };
+
+    println!("gltf: {:?}", gltf);
```
Comment (miseyu) [inline worker/crates/action-sink/src/file/gltf.rs:356]: Please delete this one!

Comment (dongri) [reply worker/crates/action-sink/src/file/gltf.rs]: deleted

File: worker/Cargo.toml (hunk 4 above)
Comment (dongri) [reply worker/Cargo.toml]: use tag

**Pull request merged**